Windows build fails

On Jenkins machine with 32-bit Cygwin Windows build fails. It does nothing special, just
```
./autogen.sh
./configure
make
```
and `make` fails with this:
```
...
...
...
   C       src/vecgf2.c => obj/vecgf2.lo
   C       src/vector.c => obj/vector.lo
   C       src/weakptr.c => obj/weakptr.lo
   C       src/gasman.c => obj/gasman.lo
   LINK    libgap.la
   LINK    gap.exe
gap.exe: initial Cygwin heap size: 2048 (0x800) MB
cp cnf/cygwin/gapicon.bmp bin/gapicon.bmp
cp cnf/cygwin/gapicon.ico bin/gapicon.ico
cp cnf/cygwin/instcygwinterminfo.sh bin/instcygwinterminfo.sh
chmod a+x bin/instcygwinterminfo.sh
uname -a > bin/cygwin-version.txt
cp .libs/cyggap-0.dll bin/i686-pc-cygwin-gcc-default32/gap.dll  # FIXME: HACK to support kernel extensions
cp: cannot create regular file â€˜bin/i686-pc-cygwin-gcc-default32/gap.dllâ€™: No such file or directory
Makefile.rules:384: recipe for target 'bin/i686-pc-cygwin-gcc-default32/gap.dll' failed
make: *** [bin/i686-pc-cygwin-gcc-default32/gap.dll] Error 1
```

This is the set of commits that triggered the failure:
* 8657f05e3e249430cc39bdc885840b05393ce683 Replaced ReadTest by Test in the dev manual
* de7b683ee2b7556ca8a58f86ef93241ceef4e402 Remove unused cnf/sysinfo.in
* 8e9e180607d014f26f9c747dc7d73f272d6ebcd8 Makefile.rules: document FORCE phony target
* 5289dbcdea846466504bc5f530ffdbaa0b8123a4 buildsys: make (re)generation of compat symlinks more robust
* 66d76a98c83b8335c111cbf574d889586fd48934 Makefile.rules: improve (dist)clean targets
* 83b3127ab4c1e68bf0d71c749ce91f06fcf062b4 Use __builtin_popcntl for counting set bits in bitlists. Can be disabledby --disable-popcnt at configure time
* e6c5462887966f769cbd679f5565f82007c6399d Update .gitignore

Those which could be relevant were added in PR #1636 and #1692 and passed on AppVeyor.

--- a/Makefile.rules
+++ b/Makefile.rules
@@ -380,7 +380,7 @@ GAP_CPPFLAGS += "-DCOMPILECYGWINDLL"
 # GAP's standard main function is renamed. And gap.exe is a tiny binary which
 # loads that DLL and calls the renamed main function.
 all: bin/$(GAPARCH)/gap.dll
-bin/$(GAPARCH)/gap.dll: libgap.la
+bin/$(GAPARCH)/gap.dll: symlinks libgap.la
 	cp .libs/cyggap-0.dll $@  # FIXME: HACK to support kernel extensions
 gap$(EXEEXT): libgap.la
 	$(QUIET_LINK)$(LINK) $(GAP_LDFLAGS) $(srcdir)/src/gapw95.c $(GAP_LIBS) libgap.la -o $@

